jupyterlab/labapp.py

```diff
@@ -897,13 +897,15 @@ def initialize(self, argv=None):
                 import jupyter_collaboration  # noqa
             except ImportError:
                 self.log.critical(
-                    """
+                    """\
+Juptyer Lab cannot start, because `jupyter_collaboration` was configured but cannot be `import`ed.
+
 To enable real-time collaboration, you must install the extension `jupyter_collaboration`.
 You can install it using pip for example:
 
   python -m pip install jupyter_collaboration
 
-This flag is now deprecated and will be removed in JupyterLab v5.
+To run without collaboration, disable collaboration (for example, remove the `--collaborative` flag from the commandline; this flag is now deprecated and will be removed in JupyterLab v5).  To see more ways to adjust the collaborative behavior, see https://jupyterlab-realtime-collaboration.readthedocs.io/en/latest/configuration.html and https://discourse.jupyter.org/t/how-to-disable-jupyter-collaboration-extension/21329/9 .
```

Comment:
The current version has a pleonasm/tautology which makes it hard to understand. It also links to the forum which we advise against.

```suggestion
If you wish to proceed without installing `jupyter_collaboration` please remove the `--collaborative` flag or switch off the `LabApp.collaborative` configuration traitlet depending on how you enabled this setting in an earlier version.

The `--colaborative` flag is not needed when `jupyter_collaboration` is installed and will be removed in JupyterLab v5. You can continue using this flag to ensure that collaborative mode is enabled, but it does not have a functional impact on `jupyter_collaboration`.

To configure or disable collaboration when `jupyter_collaboration` is installed, please refer to:
https://jupyterlab-realtime-collaboration.readthedocs.io/en/latest/configuration.html
```

I know that the Nick's message on the forum is valuable to you because it demonstrates how traitlets configuration works and that it can be provided with either a json file or on command line, however if we explained this for every single of hundreds of configuration options we have, and had a link to a post for every such a configuration option it would really be difficult to maintain, for example if someone modifies their post on the forum that we linked to. Instead what we could do is improving https://docs.jupyter.org/en/latest/use/config.html to add more examples, for example including the json file - if you would like to open a pull request there feel free to tag me for review.

Reply:
We can also improve https://jupyterlab-realtime-collaboration.readthedocs.io/en/latest/configuration.html by linking to https://docs.jupyter.org/en/latest/use/config.html and providing more examples on jupyterlab-realtime-collaboration side.

Reply:
> has a pleonasm/tautology

You mean "to run without collaboration, disable collaboration"?  It's more of an antanaclasis, right (the first "collaboration" is the python module, the second "collaboration" is the feature/concept).  Nevertheless, I see your point; I hope I've improved it now.


>  if we [had a long configuration & runtime options example] for every single of hundreds of configuration options we have [...] it would be difficult to maintain

I take your point completely -- though I think some very-common options/issues might justify this cost, but this is not one of them -- and have removed the link.


> what we could do is improving https://docs.jupyter.org/en/latest/use/config.html to add more examples, for example including the json file - if you would like to open a pull request there feel free to tag me for review.

I think this would definitely improve the situation and will give this a try sometime.